src/chain.cpp

```diff
@@ -61,6 +61,13 @@ const CBlockIndex *CChain::FindFork(const CBlockIndex *pindex) const {
     return pindex;
 }
 
+CBlockIndex *CChain::FindLatestBefore(int64_t nTime) const {
+    CBlockIndex *pindex = Tip();
```

Comment:
We could improve this since block pointers are available: `std::vector<CBlockIndex*> vChain`. See http://www.cplusplus.com/reference/algorithm/lower_bound/.